File: 41.Wave/oc/wave/wave/ViewController.m
//
//  ViewController.m
//  wave
//
//  Created by jackfrow on 2021/4/21.
//

#import "ViewController.h"
#import "ZZWave.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    NSLog(@"vmfvivhu = %.f",sinf(90));
    
    NSoperation
    

    ZZWave* wave = [[ZZWave alloc] initWithFrame:CGRectMake((([UIScreen mainScreen].bounds.size.width - 200) / 2.0), 200, 200, 200)];
    wave.color = [UIColor magentaColor];
    wave.layer.cornerRadius = 100;
    wave.layer.masksToBounds = true;
    wave.layer.borderColor = [UIColor lightGrayColor].CGColor;
    wave.layer.borderWidth = 0.5f;
    [self.view addSubview:wave];

    [wave startAnimation];
    

}


@end

File: 41.Wave/swift/ZZWave/ZZWave/ViewController.swift
//
//  ViewController.swift
//  ZZWave
//
//  Created by jackfrow on 2019/7/5.
//  Copyright © 2019 jackfrow. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var waveView : ZZWaveView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
    
        waveView = ZZWaveView(frame: CGRect(x: (UIScreen.main.bounds.width - 200)/2.0, y: 200, width: 200, height: 200))
        waveView.text = "RD"
        waveView.color = UIColor.red
        waveView.layer.cornerRadius = 100
        waveView.layer.masksToBounds = true
        waveView.layer.borderColor = UIColor.lightGray.cgColor
        waveView.layer.borderWidth = 0.5
        view.addSubview(waveView)
        
        //start.
        waveView.startAnimation()
        
    }


}

